### PR TITLE
Issue #401: Add dirty-working-tree pattern to Tier 2 catalog

### LINKED ISSUE #401: detect-wrapper-anomaly: dirty working tree (verify) パターンを Tier 2 カタログに追加
## Background

`scripts/detect-wrapper-anomaly.sh` は wrapper exit 時の既知パターンマッチを行うが、`/verify` の `Cannot run verify because there are uncommitted changes` (`VERIFY_FAILED`) パターンは未登録。Issue #393 でこのケースが発生した際、anomaly detector が空出力を返し Tier 2 catalog の自動リカバリ経路が使えず、結果として親セッションでの手動リカバリが必要だった。

## Purpose

`VERIFY_FAILED` + `uncommitted changes` シグネチャを既知パターンとして登録し、`modules/orchestration-fallbacks.md` に対応するリカバリ手順（`git status` 確認 → unrelated file なら通知して iteration retry / 関連 file なら abort）を追加する。次回以降は Tier 2 catalog で半自動リカバリ可能にする。

### Pre-merge (auto-verified)

- [x]  `scripts/detect-wrapper-anomaly.sh` に `VERIFY_FAILED` を検出するルールが追加されている
- [x]  `scripts/detect-wrapper-anomaly.sh` に `uncommitted changes` を検出するルールが追加されている
- [x]  `modules/orchestration-fallbacks.md` に dirty working tree (verify) のカタログ項目が追加されている
- [x]   カタログ項目に git status 確認フローと分岐リカバリ手順が含まれている

### Post-merge

- [ ] Issue #393 と同様の VERIFY_FAILED シナリオで、`detect-wrapper-anomaly.sh` が non-empty 出力を返し、Tier 2 catalog 経路に乗ることを確認 

## References

- Issue #393 retrospective: `docs/spec/issue-393-reconcile-review-pattern-fix.md`
- 類似 Issue: #394 (reconciler/skill ヘッダー乖離パターン), #369 (LLM-reported-success-but-no-commit パターン)

## 自動解決した曖昧ポイント

| ポイント | 選択した方針 | 理由 |
|---------|------------|------|
| Verify command 形式 | `grep -E "A\|B"` → 標準 `grep "pattern" "path"` に分割 | `-E` フラグは verify executor の標準フォーマット外。OR パターンを分割することで各文字列の存在を個別に確認する |
| AND vs OR 検出ロジック | AND（両方の文字列を個別に grep） | Purpose に「VERIFY_FAILED + uncommitted changes シグネチャ」とあり、両方が共存するケースを意図している |
| カタログ項目リカバリ手順の検証範囲 | `rubric` + `grep "git status"` を追加 | Purpose に `git status` 確認フローが明示されているが、元の AC2 はカタログ項目の存在確認のみ。内容品質を検証する条件を追加 |

### PR DESCRIPTION
**Pull request by saito**
## Summary

- `scripts/detect-wrapper-anomaly.sh` に `VERIFY_FAILED` + `uncommitted` を検出する `dirty-working-tree` パターンを追加
- `modules/orchestration-fallbacks.md` に `## dirty-working-tree` カタログ項目を追加（`git status` 確認フロー + 関連ファイル分岐リカバリ手順）
- `tests/detect-wrapper-anomaly.bats` に dirty-working-tree パターン検出のテストを3件追加

## Verification (pre-merge)

- `scripts/detect-wrapper-anomaly.sh` に `VERIFY_FAILED` を検出するルールが追加されている
- `scripts/detect-wrapper-anomaly.sh` に `uncommitted changes` を検出するルールが追加されている
- `modules/orchestration-fallbacks.md` に dirty working tree (verify) のカタログ項目が追加されている
- カタログ項目に git status 確認フローと分岐リカバリ手順が含まれている

## Verification (post-merge)

- Issue #393 と同様の VERIFY_FAILED シナリオで、`detect-wrapper-anomaly.sh` が non-empty 出力を返し、Tier 2 catalog 経路に乗ることを確認

closes #401

### CODE REVIEW

File: tests/detect-wrapper-anomaly.bats [hunk 1]
```diff
@@ -111,3 +111,26 @@ MOCK
     [ "$status" -eq 0 ]
     [ -z "$output" ]
 }
+
+@test "dirty working tree: detects VERIFY_FAILED with uncommitted changes" {
+    printf "VERIFY_FAILED\nCannot run verify because there are uncommitted changes in the working tree.\n" > "$LOG_FILE"
+    run bash "$SCRIPT" --log "$LOG_FILE" --exit-code 1 --issue 393 --phase verify
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"dirty-working-tree"* ]]
+    [[ "$output" == *"### Orchestration Anomalies"* ]]
```
Comment (saito) [inline tests/detect-wrapper-anomaly.bats:121]: **[Edge Case] tests/detect-wrapper-anomaly.bats:115-122**

SHOULD: 正常検知テストに `#393` 参照のアサーションがありません。`ANOMALY_DESC` に `Reference: #393` が必須要素として含まれているにもかかわらず（Spec 実装ステップ1）、テストでその存在を検証していないため、将来の編集で参照が削除されても検知できません。既存の他テスト（例: `pr-extraction-failure` テスト）は Issue 番号参照（`#311`）をアサートしています。

推奨修正:
```bash
@test "dirty working tree: detects VERIFY_FAILED with uncommitted changes" {
    ...
    [[ "$output" == *"dirty-working-tree"* ]]
    [[ "$output" == *"#393"* ]]
    [[ "$output" == *"### Orchestration Anomalies"* ]]
    [[ "$output" == *"### Improvement Proposals"* ]]
}
```

Severity: SHOULD

File: scripts/detect-wrapper-anomaly.sh [hunk 2]
```diff
@@ -70,6 +70,10 @@ elif grep -q "watchdog: kill and state not reached" "$LOG_FILE"; then
   PATTERN_NAME="watchdog-kill"
   ANOMALY_DESC="Watchdog killed the process in phase \`$PHASE\` (exit code $EXIT_CODE): \`watchdog: kill and state not reached\` detected. The phase did not complete within the timeout."
   IMPROVEMENT_HINT="Increase \`watchdog-timeout-seconds\` in \`.wholework.yml\` or improve liveness signals (progress output) to prevent false-positive kills. Related: #308."
+elif grep -q "VERIFY_FAILED" "$LOG_FILE" && grep -q "uncommitted" "$LOG_FILE"; then
```
Comment (saito) [inline scripts/detect-wrapper-anomaly.sh:73]: **[Edge Case] scripts/detect-wrapper-anomaly.sh:73**

CONSIDER: `dirty-working-tree` 分岐に `EXIT_CODE` ガードがありません。ログファイルが複数実行にわたって再利用された場合（前回の失敗ログが残っているケース）、`EXIT_CODE == 0` でも誤ってこのパターンが発火する可能性があります。現在の実装では `VERIFY_FAILED` は hard-error 時にのみ出力されるため実用上のリスクは低いですが、実装の意図をより明確にするために `EXIT_CODE` ガードを追加することを検討できます。

推奨修正（任意）:
```bash
elif grep -q "VERIFY_FAILED" "$LOG_FILE" && grep -q "uncommitted" "$LOG_FILE" && [[ "$EXIT_CODE" != "0" ]]; then
```

Severity: CONSIDER

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| 条件 | 結果 | 備考 |
|------|------|------|
| `scripts/detect-wrapper-anomaly.sh` に `VERIFY_FAILED` 検出ルールが追加されている | PASS | line 73 に `grep -q "VERIFY_FAILED"` 確認 |
| `scripts/detect-wrapper-anomaly.sh` に `uncommitted changes` 検出ルールが追加されている | PASS | line 73 に `grep -q "uncommitted"` 確認 |
| `modules/orchestration-fallbacks.md` に dirty working tree カタログ項目が追加されている | PASS | `## dirty-working-tree` セクション確認 |
| カタログ項目に git status 確認フローと分岐リカバリ手順が含まれている | PASS | `git status` → 関連/非関連分岐 → 通知+retry/abort 手順確認 |

## CI Status

| ジョブ名 | ステータス | 備考 |
|---------|-----------|------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | 3件の新テスト含む全テスト通過 |
| Validate skill syntax | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |
| Forbidden Expressions check | FAILURE | **PR #408 の変更とは無関係**: `docs/spec/issue-385-default-permission-mode-auto.md` の既存行に `verify hint` を検出。このファイルは PR diff に含まれていない。修正は別途対応が必要 |

## Code Review

### General Comments

特定のファイル/行に帰属できない設計・アーキテクチャレベルの問題はありません。

### Issue Count Summary

- ラインコメントとして投稿: 2 (MUST: 0, SHOULD: 1, CONSIDER: 1)
- General Comments (body): 0

---
Next: run `/merge 408`.

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| アスペクト | ファイル | 行 | 指摘 | 対応 | 修正/スキップ理由 |
|-----------|---------|-----|------|------|----------------|
| Edge Case | tests/detect-wrapper-anomaly.bats | 120 | 正常検知テストに `#393` 参照アサーションがない | 修正済み | `[[ "$output" == *"#393"* ]]` を追加 (SHOULD) |
| Edge Case | scripts/detect-wrapper-anomaly.sh | 73 | `EXIT_CODE` ガードなし | スキップ | 現実のリスクが低く、既存パターンと一貫性あり (CONSIDER) |

### Lightweight Re-check Results

- スキル構文検証: PASS (`python3 scripts/validate-skill-syntax.py skills/` → 0 error(s), 0 warning(s))
- 修正 diff チェック: PASS（アサーション1行追加のみ）
- 新規指摘: なし
